File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-java/src/main/java/com/example/statemachine/simple/doument/persist/DocumentPersistStateMachineHandler.java
package com.example.statemachine.simple.doument.persist;

import com.example.statemachine.simple.doument.DocumentEvent;
import com.example.statemachine.simple.doument.DocumentState;
import org.springframework.statemachine.StateMachine;
import org.springframework.statemachine.recipes.persist.GenericPersistStateMachineHandler;

public class DocumentPersistStateMachineHandler
  extends GenericPersistStateMachineHandler<DocumentState, DocumentEvent> {

  public DocumentPersistStateMachineHandler(
    final StateMachine<DocumentState, DocumentEvent> stateMachine) {
    super(stateMachine);
  }

}

File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-kotlin/src/main/kotlin/com/example/statemachine/simple/document/persist/DocumentPersistStateMachineHandler.kt
package com.example.statemachine.simple.document.persist

import com.example.statemachine.simple.document.DocumentEvent
import com.example.statemachine.simple.document.DocumentState
import org.springframework.statemachine.StateMachine
import org.springframework.statemachine.recipes.persist.GenericPersistStateMachineHandler

class DocumentPersistStateMachineHandler(
  stateMachine: StateMachine<DocumentState?, DocumentEvent?>?
) : GenericPersistStateMachineHandler<DocumentState?, DocumentEvent?>(stateMachine)
